### DISCUSSION TITLE
Forge WebUI Jax/Jaxlib launch error

### DISCUSSION THREAD

**Discussion by warrockmill** [lllyasviel/stable-diffusion-webui-forge#753] [Q&A]
Hello. first time posting in this so not sure if if this is the right place for this. I've been running SD Forge with no issues for SDXL generation but today have encountered an error that means I can't launch the WebUI. Could anyone help me resolve this? 

> Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
> Version: f0.0.17v1.8.0rc-latest-276-g29be1da7
> Commit hash: 29be1da7cf2b5dccfc70fbdd33eb35c56a31ffb7
> loading WD14-tagger reqs from Y:\Github\ai\Stable Diffusion Forge\webui\extensions\stable-diffusion-webui-wd14-tagger\requirements.txt
> Checking WD14-tagger requirements.
> Launching Web UI with arguments:
> Total VRAM 8192 MB, total RAM 32684 MB
> Set vram state to: NORMAL_VRAM
> Device: cuda:0 NVIDIA GeForce RTX 2060 SUPER : native
> Hint: your device supports --pin-shared-memory for potential speed improvements.
> Hint: your device supports --cuda-malloc for potential speed improvements.
> Hint: your device supports --cuda-stream for potential speed improvements.
> VAE dtype: torch.float32
> CUDA Stream Activated:  False
> 2024-05-21 16:47:39.109577: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
> 2024-05-21 16:47:39.772809: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
> Traceback (most recent call last):
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorboard\compat\__init__.py", line 42, in tf
>     from tensorboard.compat import notf  # noqa: F401
> ImportError: cannot import name 'notf' from 'tensorboard.compat' (Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorboard\compat\__init__.py)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\launch.py", line 51, in <module>
>     main()
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\launch.py", line 47, in main
>     start()
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\modules\launch_utils.py", line 541, in start
>     import webui
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\webui.py", line 19, in <module>
>     initialize.imports()
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\modules\initialize.py", line 39, in imports
>     from modules import paths, timer, import_hook, errors  # noqa: F401
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\modules\paths.py", line 60, in <module>
>     import sgm  # noqa: F401
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\repositories\generative-models\sgm\__init__.py", line 1, in <module>
>     from .models import AutoencodingEngine, DiffusionEngine
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\repositories\generative-models\sgm\models\__init__.py", line 1, in <module>
>     from .autoencoder import AutoencodingEngine
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\repositories\generative-models\sgm\models\autoencoder.py", line 12, in <module>
>     from ..modules.diffusionmodules.model import Decoder, Encoder
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\repositories\generative-models\sgm\modules\__init__.py", line 1, in <module>
>     from .encoders.modules import GeneralConditioner
>   File "Y:\Github\ai\Stable Diffusion Forge\webui\repositories\generative-models\sgm\modules\encoders\modules.py", line 5, in <module>
>     import kornia
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\kornia\__init__.py", line 11, in <module>
>     from . import augmentation, color, contrib, core, enhance, feature, io, losses, metrics, morphology, tracking, utils, x
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\kornia\x\__init__.py", line 2, in <module>
>     from .trainer import Trainer
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\kornia\x\trainer.py", line 11, in <module>
>     from accelerate import Accelerator
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\accelerate\__init__.py", line 3, in <module>
>     from .accelerator import Accelerator
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\accelerate\accelerator.py", line 41, in <module>
>     from .tracking import LOGGER_TYPE_TO_CLASS, GeneralTracker, filter_trackers
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\accelerate\tracking.py", line 43, in <module>
>     from torch.utils import tensorboard
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\torch\utils\tensorboard\__init__.py", line 12, in <module>
>     from .writer import FileWriter, SummaryWriter  # noqa: F401
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\torch\utils\tensorboard\writer.py", line 16, in <module>
>     from ._embedding import get_embedding_info, make_mat, make_sprite, make_tsv, write_pbtxt
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\torch\utils\tensorboard\_embedding.py", line 9, in <module>
>     _HAS_GFILE_JOIN = hasattr(tf.io.gfile, "join")
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorboard\lazy.py", line 65, in __getattr__
>     return getattr(load_once(self), attr_name)
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorboard\lazy.py", line 97, in wrapper
>     cache[arg] = f(arg)
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorboard\lazy.py", line 50, in load_once
>     module = load_fn()
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorboard\compat\__init__.py", line 45, in tf
>     import tensorflow
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\__init__.py", line 51, in <module>
>     from tensorflow._api.v2 import compat
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\_api\v2\compat\__init__.py", line 8, in <module>
>     from tensorflow._api.v2.compat import v1
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\_api\v2\compat\v1\__init__.py", line 30, in <module>
>     from tensorflow._api.v2.compat.v1 import compat
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\_api\v2\compat\v1\compat\__init__.py", line 8, in <module>
>     from tensorflow._api.v2.compat.v1.compat import v1
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\_api\v2\compat\v1\compat\v1\__init__.py", line 47, in <module>
>     from tensorflow._api.v2.compat.v1 import lite
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\_api\v2\compat\v1\lite\__init__.py", line 9, in <module>
>     from tensorflow._api.v2.compat.v1.lite import experimental
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\_api\v2\compat\v1\lite\experimental\__init__.py", line 8, in <module>
>     from tensorflow._api.v2.compat.v1.lite.experimental import authoring
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\_api\v2\compat\v1\lite\experimental\authoring\__init__.py", line 8, in <module>
>     from tensorflow.lite.python.authoring.authoring import compatible # line: 265
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\lite\python\authoring\authoring.py", line 43, in <module>
>     from tensorflow.lite.python import convert
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\lite\python\convert.py", line 30, in <module>
>     from tensorflow.lite.python import util
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\tensorflow\lite\python\util.py", line 51, in <module>
>     from jax import xla_computation as _xla_computation
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\jax\__init__.py", line 37, in <module>
>     import jax.core as _core
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\jax\core.py", line 18, in <module>
>     from jax._src.core import (
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\jax\_src\core.py", line 39, in <module>
>     from jax._src import dtypes
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\jax\_src\dtypes.py", line 33, in <module>
>     from jax._src import config
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\jax\_src\config.py", line 27, in <module>
>     from jax._src import lib
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\jax\_src\lib\__init__.py", line 75, in <module>
>     version = check_jaxlib_version(
>   File "Y:\Github\ai\Stable Diffusion Forge\system\python\lib\site-packages\jax\_src\lib\__init__.py", line 67, in check_jaxlib_version
>     raise RuntimeError(
> RuntimeError: jaxlib version 0.4.28 is newer than and incompatible with jax version 0.4.26. Please update your jax and/or jaxlib packages.
> Press any key to continue . . .

I've tried updating and saw a similar discussion on updating Jax but I've had no luck in doing that. This happened after I updated some extensions. Adetailer and Forge Couple.

**Comment by gltchcom** [comment]
It happened to me when I updated adetailer. I'm not sure how to fix this.

**Comment by klintan96** [comment]
delete the adetailer folder from the extensions folder
delete the system folder from the root folder of SD Forge
re-download Forge https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/latest/webui_forge_cu121_torch21.7z
еxtract only the system folder from the archive
run.bat
install adetailer

**Reply by ExissNA** [reply]
Saved me a lot of headache.  Usually when I update things there isn't issue - this had me back up and running in only the amount of time it took to copy the files and let Forge perform initial setup again.  adetailer has volunteered itself for the chopping block, I never used it on Forge anyways.  

Thank you.
